### PR TITLE
Added `publish` command `--noCommit` option

### PR DESCRIPTION
**Pull request by jakubmeysner**
Resolves microsoft/msstore-cli#32

### CODE REVIEW

**Comment by jakubmeysner** [comment]
@microsoft-github-policy-service agree

**Comment by jakubmeysner** [comment]
@azchohfi hi, I've rebased the branch based on `main`.

File: MSStore.CLI/Commands/PublishCommand.cs [hunk 1]
```diff
@@ -50,10 +54,25 @@ public PublishCommand()
...
+                },
                 description: "Specifies the Application Id. Only needed if the project has not been initialized before with the 'init' command.");
 
             AddOption(appIdOption);
+
+            var noCommitOption = new Option<bool>(
+                aliases: new string[]
+                {
+                    "--noCommit",
+                    "-nc"
```
Comment (azchohfi) [inline MSStore.CLI/Commands/PublishCommand.cs:70]: Why "nd"? Does "nc" makes more sense?

Comment (jakubmeysner) [reply MSStore.CLI/Commands/PublishCommand.cs:70]: @azchohfi fixed in a previous update.

**Comment by jakubmeysner** [comment]
@azchohfi hi, the issues that were causing build failures should be resolved now, can you re-approve the workflow run?

**Comment by jakubmeysner** [comment]
@azchohfi it seems to be a token permissions issue: https://github.com/dorny/test-reporter?tab=readme-ov-file#recommended-setup-for-public-repositories

**Comment by azchohfi** [comment]
Please, merge main into your branch and we should be good to go.

**Comment by azchohfi** [comment]
Thank you for your contribution!

**Pull request merged**